### PR TITLE
search: add button to export results

### PR DESCRIPTION
**Pull request by lauren-d**
This part add a button or dropdown list on search page to export results.

* Adapts record search component and template.
* Adds `export as ...` on ng-core-tester home page.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>
Co-Authored-by: Lauren-D <laurent.dubois@itld-solutions.be>

## Why are you opening this PR?

- Which task/US does it implement?
https://tree.taiga.io/project/rero21-reroils/task/1616

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.html [hunk 1]
```diff
@@ -29,7 +29,7 @@
 <!-- each object properties -->
 <div class="{{ getCssClass() }}">
   <ng-container *ngFor="let f of field.fieldGroup">
-    <div class="{{ getCssClass(f) }}" *ngIf="!f.hide" [id]="f.id">
+    <div class="{{ getCssClass(f) }}" *ngIf="!f.hide">
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.html:32]: I know it's for correcting a bug, but are you sure removing the ID does not break the functionality of the commit https://github.com/rero/ng-core/commit/87e51ef573d4e49c4921045b3780602a81b5caf3 ?

File: projects/rero/ng-core/src/lib/record/search/record-search.component.html [hunk 2]
```diff
@@ -54,6 +54,32 @@ <h5 *ngIf="types.length == 1 && showLabel">
...
+           [href]="getExportFormatUrl(exportFormats[0].format)" >
+          <i class="fa fa-download mr-1"></i> {{ 'Export as' | translate }} {{ exportFormats[0].label }}
+        </a>
+        <div class="btn-group ml-2" dropdown *ngIf="exportFormats && exportFormats.length > 1">
+          <button
+            id="button-export-basic"
+            dropdownToggle
+            type="button"
+            class="btn btn-outline-primary dropdown-toggle"
+            aria-controls="dropdown-download-basic"
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.html:67]: Not sure this property is useful.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.html [hunk 3]
```diff
@@ -54,6 +54,32 @@ <h5 *ngIf="types.length == 1 && showLabel">
...
+          >
+            <i class="fa fa-download mr-1"></i> {{ 'Export as' | translate }} ... <span class="caret"></span>
+          </button>
+          <ul
+            id="dropdown-export-basic"
+            *dropdownMenu
+            class="dropdown-menu"
+            role="menu"
+            aria-labelledby="button-basic"
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.html:76]: This attribute value does not correspond to button ID.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 4]
```diff
@@ -177,7 +178,8 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
     private _recordUiService: RecordUiService,
     private _recordSearchService: RecordSearchService,
     private _translateService: TranslateService,
-    private _spinner: NgxSpinnerService
+    private _spinner: NgxSpinnerService,
+    private _apiService: ApiService
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:182]: Add this param to doc block.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 5]
```diff
@@ -454,6 +456,35 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
     return null;
   }
 
+  /**
+   * Export formats.
+   * @return Array of export format or an empty array.
+   */
+  get exportFormats() {
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:463]: Add a return type `: Array<any>`

File: projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.html (hunk 1 above)
Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.html:32]: I checked in rero-ils-ui editor and it seems that doesn't break the previous fix. 
The `jump to` seems to works properly but a double check must be done to confirm that.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts (hunk 4 above)
Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.ts:183]: Done

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts (hunk 5 above)
Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.ts:464]: Done

File: projects/rero/ng-core/src/lib/record/search/record-search.component.html (hunk 3 above)
Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.html]: Ok

File: projects/rero/ng-core/src/lib/record/search/record-search.component.html (hunk 2 above)
Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.html]: removed

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 6]
```diff
@@ -454,6 +457,35 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
     return null;
   }
 
+  /**
+   * Export formats.
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:461]: This documentation seems a bit useless, can you mention at least that this values comes from the configuration?

Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.ts]: Ok done.

File: projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.html (hunk 1 above)
Comment (sebastiendeleze) [reply projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.html:32]: @zannkukai Can you confirm that removing `id` attribute does not break your changes?

Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.html]: changes reversed

**Pull request merged**